src/marqo/core/index_management/vespa_application_package.py

```diff
@@ -86,7 +86,7 @@ def _cleanup_container_config(self):
         """
         container_element = self._ensure_only_one('container')
         for child in container_element.findall('*'):
-            if child.tag in ['document-api', 'search']:
+            if child.tag in ['document-api', 'document-processing', 'search']:
                 child.clear()
```

Comment:
Why clear the element if we are not regenerating it? Shouldn't we just keep it unchanged?

Reply:
> Shouldn't we just leave it unchanged?

It seems we are doing destructive changes here. But the logic behind it is the same as deleting all other components, to make sure all config is controlled by Marqo. For Cloud, it should be the same since we don't config document-processing anyways. It wouldn't be more risky for OS users either since the previous logic is to delete this element.  On the other hand, clearing it makes it simple to change its config in the future since we do not need complicated logic to check the existing config (random) and decide what to keep and what to override.